README.md

```diff
@@ -0,0 +1,158 @@
+
+The way it works is by running a set of scripts that interact with the hardware using vendor tools/other methods while exposing a fake RedFish API that Metal3 can query.
+
+The [app/](./app/) directory contains the FakeFish application. Inside the `app` directory we can find the [custom_scripts](./app/custom_scripts/) folder where we will find 5 scripts:
+
+|Script|What should it do?|
+|------|----------------|
+|`poweron.sh`|Power on the server|
+|`poweroff.sh`|Power off the server|
+|`bootfromcdonce.sh`|Set server to boot from virtual CD once|
+|`mountcd.sh`|Mount the iso received in the server's virtual CD|
+|`unmountcd.sh`|Unmount the iso from the server's virtual CD|
+
+The script names can't be changed, and the scripts in the `custom_scripts` folder are just placeholders. Users need to implement their own scripts, we will not maintain/add providers to this project.
+
+Users can take a look at the [dell_scripts](./dell_scripts/) folder where they can find the scripts we used for our testing with DELL hardware. Remember that these scripts are unsupported/unmaintained and should be taken as a reference, nothing else.
+
+A [Containerfile](./app/Containerfile) is included, so users can build their own container image out of it. We published an image using the bits of the first release at `quay.io/mavazque/fakefish:v0`.
+
+## Usage
+
+You need a FakeFish process for each server you plan to install. Think of FakeFish like if it was a custom implementation of a BMC for that specific server.
+
+Since you will be potentially running multiple FakeFish instances, you will make use of an environment variable to configure in which port a given FakeFish instance listens on. On top of that, you need to do a bind mount for the folder containing the scripts for managing that specific server.
+
+An example can be found below:
+
+> **NOTE**: Every container is mapped to a single BMC, but if more hosts are required, different ports can be used (9001, 9002,...)
+
+```sh
+podman run -p 9000:9000 -e PORT=9000 -v $PWD/dell_scripts:/opt/fakefish/custom_scripts:z quay.io/mavazque/fakefish:v0
+
+sudo firewall-cmd --add-port=9000/tcp
+```
+
+Then, in the `install-config.yaml` file, it is required to specify the IP where the container is running instead of the 'real' BMC:
+
+```yaml
+bmcAddress: redfish-virtualmedia://192.168.1.10:9000/redfish/v1/Systems/1
+```
+
+## Logs
+
+In a successful execution you should see something like this in the logs:
+
+- Starting FakeFish
+
+    ```sh
+    $ podman run -p 9000:9000 -e PORT=9000 -v $PWD/dell_scripts:/opt/fakefish/custom_scripts:z quay.io/mavazque/fakefish:v0
+
+     * Serving Flask app 'fakefish' (lazy loading)
+     * Environment: production
+       WARNING: This is a development server. Do not use it in a production deployment.
+       Use a production WSGI server instead.
+     * Debug mode: off
+     * Running on all addresses.
+       WARNING: This is a development server. Do not use it in a production deployment.
```

Comment:
This is also something we need to address before we start offering this as a customization option. Probably by using gunicorn/uwsgi.